.github/workflows/release.yaml

```diff
@@ -0,0 +1,93 @@
+name: Release Tool
+
+on:
+  workflow_dispatch:
+    inputs:
+      step:
+        description: 'Which step do you want to (re-)run?'
+        required: true
+        type: choice
+        options:
+          - 2-prepare-release
+          - 4-post-release
+      version:
+        description: 'Which version are you releasing? (e.g. vX.Y.Z[-(pre|rc).W])'
+        required: true
+        type: string
+        default: vX.Y.Z
+
+permissions:
+  # To be able to access the repository with `actions/checkout`
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.inputs.version }}
+  cancel-in-progress: true
+
+jobs:
+  release:
+    name: Release
+    environment: release-tool
+    timeout-minutes: 10
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Install Go
+        if: ${{ steps.cache.outputs.cache-hit != 'true' }}
+        uses: actions/setup-go@cdcb36043654635271a94b9a6d1392de5bb323a7 # v5.0.1
+        with:
+          # renovate: datasource=golang-version depName=go
+          go-version: 1.22.4
+
+      - name: Get token
+        id: get_token
+        uses: cilium/actions-app-token@61a6271ce92ba02f49bf81c755685d59fb25a59a # v0.21.1
+        with:
+          APP_PEM: ${{ secrets.CILIUM_RELEASE_BOT_PEM }}
+          APP_ID: ${{ secrets.CILIUM_RELEASE_BOT_APP_ID }}
+
+      - name: Authenticate with GH CLI
+        run: |
+          gh auth login --with-token <<< "${{ steps.get_token.outputs.app_token }}"
+
+      - name: Checkout release tool
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+        with:
+          persist-credentials: false
+          repository: cilium/release
+          path: "./release"
+
+      - name: Move release source code to upper directory
+        run: mv release ../
+
+      - name: Checkout cilium source code
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+        with:
+          ref: ${{ github.event.repository.default_branch }}
```

Comment:
Does this mean that the release is always based on the tip of the `main` branch?

Reply:
Yes, should we pin it and let renovate update it automatically?

Reply:
Maybe I misunderstand.

I thought the intent for this workflow is that you can ask it to release an RC on main, or a patch release on a branch, but it seems like this line always checks out the tip of main, so the workflow will tag the release on this commit reference.

If we're only supporting prereleases for now that's fine. Or if this line of code is not actually what is choosing the target commit for the release, that's also fine. I don't understand how this could be pinned to some version and get updated by renovate.